client/www/components/ui.tsx

```diff
@@ -1077,6 +1077,48 @@ export const InfoTip = ({ children }: PropsWithChildren) => {
   );
 };
 
+export function ProgressButton({
+  percentage = 0,
+  loading,
+  className,
+  children,
+  variant,
+  ...props
+}: PropsWithChildren<{
+  percentage?: number;
+  loading?: boolean;
+  className?: string;
+  variant?: 'primary' | 'secondary' | 'subtle' | 'destructive' | 'cta';
+}> &
+  Parameters<typeof Button>[0]) {
+  const progressFillStyle = {
+    width: loading ? `${Math.max(0, Math.min(100, percentage))}%` : '0%',
```

Comment:
If I remember correctly `width` can cause layout reflows in the browser. Maybe transform: scaleX will make this faster?